docs/integrations/sources/xero.md

```diff
@@ -0,0 +1,58 @@
+# Airbyte Source Connector for Xero
+
+This is a setup guide for the Xero source connector which ingests data from the Accounting API.
+
+## Prerequisites
+
+First of all you should create an application in [Xero development center](https://developer.xero.com/app/manage/). The only supported integration type is to use [Xero Custom Connections](https://developer.xero.com/documentation/guides/oauth2/custom-connections/developer) so you should choose it on creating your Xero App. 
```

Comment:
@alkali could you provide some context as to why this requires a Xero Custom Connection?

That's quite a limitation for some use cases for this connector, I would imagine.

Would someone be able to pick this up and adapt your code so that it could work with some extra short-lived token / refresh token logic?

Reply:
@stefl Xero `access_token` lives [30 minutes](https://developer.xero.com/documentation/guides/oauth2/auth-flow/#refreshing-access-and-refresh-tokens). `Refresh_token` lives 60 days, but if an application has used it for updating `access_token`, Xero will give you a new `refresh_token` and the old one will expire 30 minutes after usage. So it basically means that after refreshing `access_token` we should save a new `refresh_token` somewhere in the credentials (in the connector's config). This is impossible for now and I was waiting for https://github.com/airbytehq/airbyte/issues/3990 to be resolved.

I have a workaround (I am actually using this Xero connector at my job for a while) - use an additional custom OAuth2Connector, which sends a callback with a new `refresh_token` to a particular URL, and some code at this particular URL updates the connector's config with new `refresh_token`.  But I'm pretty sure this is not the way for a publicly released connector.